### ISSUE TITLE
Have constant values on predict

### ISSUE THREAD

**Issue by mmaxxi** [tiny-dnn/tiny-dnn#30]
Hello!
I have such a problem: when I'm trying to predict class, I get the same value vector, no matter, what input data there is.
As input the net has 32x32 image. As output - two classes, or simplier - just yes/no.
So, I always get output answer smth like this: [0.8, -0.8].
After training I get a file which looks like good (many different numbers).
What could I forget?
Thank you.

Source code:

``` c++

// NeuralNetwork::fNet has type network<mse, adagrad>

void NeuralNetwork::CreateNet()
{
    fNet << convolutional_layer<tan_h>(32, 32, 5, 1, 6) // C1, 1@32x32-in, 6@28x28-out
       << average_pooling_layer<tan_h>(28, 28, 6, 2) // S2, 6@28x28-in, 6@14x14-out
       << convolutional_layer<tan_h>(14, 14, 5, 6, 16) // C3, 6@14x14-in, 16@10x10-in
       << average_pooling_layer<tan_h>(10, 10, 16, 2) // S4, 16@10x10-in, 16@5x5-out
       << convolutional_layer<tan_h>(5, 5, 5, 16, 120) // C5, 16@5x5-in, 120@1x1-out
       << fully_connected_layer<tan_h>(120, 2); // F6, 120-in, 2-out
}

void NeuralNetwork::Train(string trainDataFilename, string neuralNetFilename)
{
    CreateNet(32);

    vector<label_t> Y;
    vector<vector<double>> X;

        // This function is OK 100%
    ReadTrainData(trainDataFilename, X, Y);

    // create callback
    auto on_enumerate_epoch = [&](){
        nEpoch++;

        SYSTEMTIME now;
        GetSystemTime(&now);
        cout << endl << "Finished epoch #" << nEpoch << " - " << now.wHour << ":" << now.wMinute << endl;
    };

    auto on_enumerate_minibatch = [&](){
        cout << ".";
    };

    int minibatch_size = 10;
    int num_epochs = 30;
    fNet.optimizer().alpha *= std::sqrt(minibatch_size);
    fNet.train(X, Y, minibatch_size, num_epochs, on_enumerate_minibatch, on_enumerate_epoch);

    ofstream os(neuralNetFilename); 
    fNet.save(os);
    os.close();

    FilePath = neuralNetFilename;
}

void NeuralNetwork::LoadFromFile(string netFilename)    
{
    CreateNet(32);
    if (!netFilename.empty())
    {
        ifstream ifs(netFilename);
        if (ifs.is_open())
        {
            fNet.load(ifs);
            ifs.close();
        }
    }
}

label_t NeuralNetwork::Predict(vector<double>& x)
{
    if (IsReady())
    {
        auto y = fNet.predict(x);

        double maxval = -1000;
        label_t maxi = -1;
        for (int i = 0; i < y.size(); ++i)
        {
            if (y[i] > maxval)
            {
                maxval = y[i];
                maxi = i;
            }
        }
        return maxi;
    } else {
        return -1000000;
    }
}
```

**Comment by mmaxxi** [comment]
Ok, I understood, that I should make rescaling, that 0.8=tanh(1).
But values are constant for pos and neg input.
Training takes too much time, so I tried it on 5 epoches. Can this be a problem? Just don't want to waste about day or two on 30 epoches...

**Comment by micahpearlman** [comment]
I'm getting the same issue with roughly the same code (I can post my exact code if needed).  Any resolution?

**Comment by micahpearlman** [comment]
Turns out that for the training I needed to randomize the labeled images.  For example I had 10000 images of label 0 and 10000 images of label 1 all ordered one label after another in the vector of images.  I randomized the order and now the training is working as well as the prediction.  Let me know if you need further clarification on this.

**Comment by mmaxxi** [comment]
Oh, thank you for an idea, I'll try it.
I think, this is a solution.

**Comment by edgarriba** [comment]
@mmaxxi Solved that?

**Comment by Randl** [comment]
@mmaxxi Do you still have any problems?

**Comment by Randl** [comment]
No reply, closing
